### PR TITLE
Fix voronoi and gradient noise when using half precision

### PR DESCRIPTION
**Pull request by rjonaitis**
### Purpose of this PR
Fix shadergraph's voronoi and gradient noise nodes producing artifacts when used with half precision on mobiles (case 1232018).
example

---
### Testing status

**Manual Tests**: What did you do?
Manually tested affected project on Android, Mali and Adreno GPUs

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
voronoimaster
Voronoi on master
voronoiafterfix
Voronoi on this branch

Do we want to include a fix that changes the rng algorithm of the generated voronoi? I feel like this will cause regressions in people's shaders, with voronoi that looked good in a certain situation changing and causing it to no longer look good (for instance, voronoi that was tweaked to have a cell at a specific position, now no longer has a cell there)

Is it possible to wrap the change to the algorithm in ifdefs so it is only changed in the edge case where it needs to be?

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/Noise/VoronoiNode.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public override void GenerateNodeFunction(FunctionRegistry registry, GenerationM
 inline $precision2 Unity_Voronoi_RandomVector_$precision ($precision2 UV, $precision offset)
 {
     $precision2x2 m = $precision2x2(15.27, 47.63, 99.41, 89.98);
-    UV = frac(sin(mul(UV, m)) * 46839.32);
```
Comment (alindmanUnity) [inline com.unity.shadergraph/Editor/Data/Nodes/Procedural/Noise/VoronoiNode.cs:61]: can we simply change this number to be `46839` to try and maintain more consistent behavior?

**Comment by rjonaitis** [comment]
the multiplication by `46839.32` causes artifacts even when the Voronoi node is set to use full precision, at least on S10e (Mali-G76). I've asked the original author about this magic number, his answer was: "Its not needed.  We did this during hack week several years ago and I was just playing with the numbers to get a pattern that looked nice for our demo."
`Is it possible to wrap the change to the algorithm in ifdefs so it is only changed in the edge case where it needs to be?`
I can't see any way how to do that, as it is not an edge case, it's a general case. Probably the solution would be to expose this rng seed as voronoi node input with default value of  `46839.32`, so that it would not regress, and provide possibility for users to change it if it causes problems.
voronoi_bad_mali

File: com.unity.shadergraph/Editor/Data/Nodes/Procedural/Noise/VoronoiNode.cs (hunk 1 above)
Comment (rjonaitis) [reply com.unity.shadergraph/Editor/Data/Nodes/Procedural/Noise/VoronoiNode.cs:61]: no, removing just the fractional part does not fix the issue

**Review by alindmanUnity — APPROVED** [review]
change looks good to me, if QA thinks that the behavior change is too drastic then please add the seed as an exposed slot with the default value to the previous seed value to prevent this

**Review by ghost — APPROVED** [review]
Talked it over with Marc Templin and I don't think the change is too drastic. The vast majority of users won't notice and it's only a change between major versions, so I'm going to approve.

**Pull request merged**